src/Appwrite/Utopia/Response/Model/ResourceToken.php

```diff
@@ -61,24 +61,40 @@ public function __construct()
 
     public function filter(Document $document): Document
     {
-        $maxAge = PHP_INT_MAX;
         $expire = $document->getAttribute('expire');
-
+        $now = new \DateTime();
+        
+        // Calculate expiration timestamp for JWT
+        $expTimestamp = null;
         if ($expire !== null) {
-            $now = new \DateTime();
             $expiryDate = new \DateTime($expire);
-
-            // set 1 min if expired, we check for expiry later on route hooks for validation!
-            $maxAge = min(60, $expiryDate->getTimestamp() - $now->getTimestamp());
+            $secondsUntilExpiry = $expiryDate->getTimestamp() - $now->getTimestamp();
+            
+            // If token is expired, set expiration to 1 minute from now
+            // We check for actual expiry later on route hooks for validation
+            if ($secondsUntilExpiry <= 0) {
+                $expTimestamp = $now->getTimestamp() + 60;
+            } else {
+                $expTimestamp = $expiryDate->getTimestamp();
+            }
```

Comment:
This seems excessive given we already have `$expire`.

Also, let's get rid of that 1 minute expiration adjustment. I'm pretty sure it was a hack because maxAge had to be greater than 0.